File: Darts/src/code/reporting/ParticipantWrapper.java
package code.reporting;

/**
 * Lightweight wrapper object to represent a participant
 * Used in reporting
 */
public final class ParticipantWrapper
{
	private String playerName = null;
	private int finishingPosition = -1;
	
	public ParticipantWrapper(String playerName, int finishingPosition)
	{
		this.playerName = playerName;
		this.finishingPosition = finishingPosition;
	}
	
	@Override
	public String toString()
	{
		return playerName + " (" + getPositionDesc() + ")";
	}
	private String getPositionDesc()
	{
		if (finishingPosition == -1)
		{
			return "-";
		}
		
		return "" + finishingPosition;
	}
	
	public String getPlayerName()
	{
		return playerName;
	}
	public int getFinishingPosition()
	{
		return finishingPosition;
	}

}

File: Darts/src/burlton/dartzee/code/reporting/ParticipantWrapper.kt
package burlton.dartzee.code.reporting

/**
 * Lightweight wrapper object to represent a participant
 * Used in reporting
 */
class ParticipantWrapper(private val playerName: String, val finishingPosition: Int)
{
    override fun toString() = "$playerName (${getPositionDesc()})"

    private fun getPositionDesc() = if (finishingPosition == -1) "-" else "$finishingPosition"
}